src/test/java/com/depli/controller/JMXNodeControllerTest.java

```diff
@@ -0,0 +1,158 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import io.github.benas.randombeans.api.EnhancedRandom;
+
+import static org.mockito.ArgumentMatchers.any;
+
+/**
+ * Created by sfatali on October 1, 2017.
+ */
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = DepliApplication.class)
+@SpringBootTest
+public class JMXNodeControllerTest {
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @MockBean
+    private JMXNodeService jmxNodeService;
+
+    private MockMvc mockMvc;
+
+    private JMXNode jmxNode;
+    private List<JMXNode> jmxNodeList = new ArrayList<>();
+    private EnhancedRandom enhancedRandom;
+    private ObjectMapper mapper;
+
+    @Before
+    public void setup() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+
+        enhancedRandom = EnhancedRandomBuilder.aNewEnhancedRandomBuilder().build(); // randomizer
+        jmxNode = enhancedRandom.nextObject(JMXNode.class); // creating random node
+        jmxNodeList.add(jmxNode); // adding random node to list of all nodes
+        mapper = new ObjectMapper(); // needed for POST and PUT
+    }
+
+    @Test
+    public void getAll() throws Exception {
+        Mockito.when(jmxNodeService.findAll()).
+                thenReturn(jmxNodeList);
+
+        RequestBuilder requestBuilder = MockMvcRequestBuilders.get("/nodes")
+                .accept(MediaType.APPLICATION_JSON);
+        mockMvc.perform(requestBuilder).andExpect(status().isOk())
+                .andExpect(jsonPath("$").isNotEmpty())
+                .andExpect(jsonPath("$").isArray())
+                .andDo(print());
+    }
+
+    @Test
+    public void getByNodeId() throws Exception {
+        Mockito.when(jmxNodeService.findByNodeId(jmxNode.getNodeId()))
+                .thenReturn(jmxNode);
+
+        // success case
+        RequestBuilder requestBuilderForSuccess = MockMvcRequestBuilders.get("/nodes/"+jmxNode.getNodeId())
+                .accept(MediaType.APPLICATION_JSON);
+        mockMvc.perform(requestBuilderForSuccess).andExpect(status().isOk())
+                .andExpect(jsonPath("$").isNotEmpty())
+                .andExpect(jsonPath("$.nodeName").value(jmxNode.getNodeName()))
+                .andDo(print());
+
+        //error case
+        RequestBuilder requestBuilderForError = MockMvcRequestBuilders.get("/nodes/"+Math.random())
+                .accept(MediaType.APPLICATION_JSON);
+        mockMvc.perform(requestBuilderForError).andExpect(status().isBadRequest())
+                .andExpect(jsonPath("$.error").exists())
+                .andDo(print());
+    }
+
+    @Test
+    public void saveNewNode() throws Exception {
+        // success case - does not work
+        Mockito.when(jmxNodeService.save(any()))
+                .thenReturn(true); // mockito does not return true here, I don't know why! :((
```

Comment:
It would be nice if you can debug and identify what is wrong there. Also please keep out this kind of comments from the code. If you want to say something, comment it like this. :-)